### ISSUE TITLE
Enhancement: [no-unused-constructor-member] Should report unused members in constructor

### ISSUE THREAD

**Issue by muuvmuuv** [typescript-eslint/typescript-eslint#6053]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

There is none yet?

### Description

Should report a warning when a private member in a constructor is unused.

### Fail

```typescript
class Test {
  constructor(private foo: Foo) {}
}
```

### Pass

```typescript
class Test {
  constructor(private foo: Foo) {}

  test() {
    return this.foo;
  }
}
```

### Additional Info

Maybe as well for non-private, not declared as public.

**Comment by JoshuaKGoldberg** [comment]
Thanks for posting! This is a duplicate of #4571. Before filing an issue, please [use our issue search](https://github.com/typescript-eslint/typescript-eslint/issues) to check for open and closed issues that already address what you're looking for.

**Comment by muuvmuuv** [comment]
Oh ok, I thought this issue is just for private class members, my proposal is for constructor members/definitions.

**Comment by JoshuaKGoldberg** [comment]
Ah I can see why you'd suggest a new issue. Constructor parameter properties with the `private` modifier _are_ private class members though.

**Comment by muuvmuuv** [comment]
Ohh, sorry, didn't know, thought those are handled different. 👍🏻 hope both are addressed the same in that issue, really think this would lets us find a lot of unused props in our codebase.
